src/cocotb_bus/monitors/__init__.py

```diff
@@ -108,7 +111,7 @@ async def wait_for_recv(self, timeout=None):
         else:
             await self._wait_event.wait()
 
-        return self._wait_event.data
+        return self._wait_event_data
```

Comment:
Is this correct?

Reply:
Thanks for noticing this.

Sorry for not double checking the state of the PR. This ended up needing a more complicated solution because the event data may be needed by the callbacks, accordingly it _must_ be stored by `Event.set()` on 1.9.x. Setting it before `Event.set()` doesn't work because it will then be overwritten, setting it after `Event.set()` doesn't work because then callbacks won't see the updated value.

The PR has been updated with the fix.